pennylane/tape/qscript.py

```diff
@@ -428,10 +428,12 @@ def _update_par_info(self):
                 {"op": op, "op_idx": idx, "p_idx": i} for i, d in enumerate(op.data)
             )
 
+        n_ops = len(self.operations)
         for idx, m in enumerate(self.measurements):
             if m.obs is not None:
                 self._par_info.extend(
-                    {"op": m.obs, "op_idx": idx, "p_idx": i} for i, d in enumerate(m.obs.data)
+                    {"op": m.obs, "op_idx": idx + n_ops, "p_idx": i}
```

Comment:
can you confirm what we learned last time we looked at this, and that there's nothing else that needs changing? just pulse_grad and hadamard_grad would maybe care, but they don't concern themselves with measurements anyway right?

Reply:
As far as I can see, any gradient transforms that use `get_operation` either only care about the operation and not the index, or only use the index to access `tape.operations`. In either case, having items in `_par_info` with indices that exceed `len(tape.operations)` should be fine.